addons/storysource/src/loader/generate-helpers.js

```diff
@@ -27,14 +27,21 @@ function generateSourceWithoutUglyComments(source, { comments, uglyCommentsRegex
   return parts.join('');
 }
 
-function prettifyCode(source, { prettierConfig, parser }) {
+function prettifyCode(source, { prettierConfig, parser, filepath }) {
   let config = prettierConfig;
 
-  if (!config.parser && parser && parser !== 'javascript') {
-    config = {
-      ...prettierConfig,
-      parser,
-    };
+  if (!config.parser) {
+    if (parser) {
+      config = {
+        ...prettierConfig,
+        parser: parser === 'javascript' ? 'babylon' : parser,
```

Comment:
Not needed anymore

Reply:
I thought the same.
But in [this test](https://github.com/storybooks/storybook/blob/master/addons/storysource/src/loader/inject-decorator.test.js) , if the extension is txt and the contents are something else, prettier does not support to txt file, so it gets an error.

error:
```
No parser could be inferred for file: (LOCAL_PATH)/addons/storysource/src/loader/__mocks__/inject-decorator.stories.txt
```

Should i change the test?
I think that probably this is a rare case, so I think that do not have to deal with it unless  need it.

Reply:
After giving it a second thought, we probably should always pass parser to prettier when user specifies it explicitly, which is exactly what we do now. So let's keep it as is

Reply:
But we probably should put `if (filepath)` inside `if (!config.parser)`, to cover cases when parser is not passed

Reply:
Or maybe we should just pass `babylon` when `parser` isn't set explicitly, because this is what we use by default ourselves. In this case, `filepath` will be never needed